sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -582,7 +582,10 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
   override def third: Expression = keyValueDelim
 
   override def inputTypes: Seq[AbstractDataType] =
-    Seq(StringTypeNonCSAICollation, StringTypeNonCSAICollation, StringTypeNonCSAICollation)
+    Seq(
+      StringTypeNonCSAICollation(supportsTrimCollation = true),
+      StringTypeNonCSAICollation(supportsTrimCollation = true),
+      StringTypeNonCSAICollation(supportsTrimCollation = true))
```

Comment:
Sorry for the confusion, latest update for this is that we will disable collations support for all expressions that are using regexes in them. Since `str_to_map` uses regex split inside its implementation, it will not be enabled even for `UTF8_BINARY_TRIM` collation. Therefore, please remove the changes for `str_to_map` expression, and then I will approve the PR. Btw, the disablement will be done in a separate PR, you do not have to do anything else than reverting changes for `str_to_map`.